File: 1758. Minimum Changes To Make Alternating Binary String.py
class Solution(object):
    def minOperations(self, s):
        count = 0 # for when it starts with 0

        for i in range(len(s)):
            if i % 2 == 0:
                if s[i] != '0':
                    count +=1
            else:
                if s[i] != '1':
                    count +=1
        
        return min(count, len(s) - count)

File: 1758. Minimum Changes To Make Alternating Binary String.rs
impl Solution {
    pub fn min_operations(s: String) -> i32 {
        let mut res = 0;
        let mut bit = 0u8;
        for c in s.bytes() {
            res += ((c - b'0') ^ bit) as i32;
            bit ^= 1;
        }
        return res.min(s.len() as i32 - res);
    }
}
